assets/user-profile/components/PauseNotificationModal.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IUser} from 'interfaces';
+import {gettext} from 'utils';
+import {modalFormInvalid, modalFormValid} from 'actions';
+import {updateUserNotificationPause} from 'user-profile/actions';
+import Modal from 'components/Modal';
+
+interface IProps {
+    modalFormInvalid(): void;
+    modalFormValid(): void;
+    updateUserNotificationPause(schedule: Omit<IUser['notification_schedule'], 'last_run_time'>): void;
+    data: {
+        user: IUser;
+    };
+}
+
+interface IState {
+    pauseFrom?: string;
+    pauseTo?: string;
+}
+
+const disabledOptionDateFrom = new Date().toISOString().split('T')[0];
+
+const ONE_DAY = 1; // representing one day, can be used for adding or subtracting a day depending on the context of the function
+
+class PauseNotificationModalComponent extends React.Component<IProps, IState> {
+    formRef: React.RefObject<HTMLFormElement>;
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            pauseFrom: this.props.data.user.notification_schedule?.pauseFrom ?? '',
+            pauseTo: this.props.data.user.notification_schedule?.pauseTo ?? '',
+        };
+
+        this.onSubmitForm = this.onSubmitForm.bind(this);
+        this.updateDate = this.updateDate.bind(this);
+        this.formRef = React.createRef<HTMLFormElement>();
+    }
+
+    componentDidMount() {
+        this.props.modalFormValid();
+    }
+
+    onSubmitForm(event: React.FormEvent<HTMLFormElement>) {
+        event.stopPropagation();
+        event.preventDefault();
+
+        if (this.formRef.current == null) {
+            throw new Error('ref missing');
+        }
+
+        const inputs = this.formRef.current.querySelectorAll('input') as NodeListOf<HTMLInputElement>;
+
+        const hasErrors = [...inputs].some(input => {
+            return input.checkValidity() !== true;
+        });
+
+        if (hasErrors === true) {
+            this.formRef.current.reportValidity();
+        } else {
+            this.props.updateUserNotificationPause(this.state);
+        }
+    }
+
+    updateDate(event: any, state: 'pauseFrom' | 'pauseTo') {
```

Comment:
don't call the argument `state` not to confuse it with component state.

Also `'pauseFrom' | 'pauseTo'` isn't a state of anything so naming is definitelly off.

I see you use `any` type here as well - go over the entire PR and ensure you are not using `any` in your newly written code.